algviz/parser/structures.py

```diff
@@ -213,11 +214,33 @@ class TreeNode(DataStructure):
     """A node with some number of children in a fixed order.  Edges are implicit."""
     # A common superclass could be used for linked-list nodes, since linked
     # lists are just skinny trees
+
     def __init__(self, data, children=None, **kwargs):
         super().__init__(**kwargs)
         self.data = data
         self.children = [] if children is None else children
 
+    def is_leaf(self):
+        return len(self.children) == 0
+        
+    def tree_width(self):
```

Comment:
Method name should just be `width`.  Or make it a standalone function called `tree_width`, defined closer to where it is used.